java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,656 @@
+        ArrayAccessExpressionTree aaet = (ArrayAccessExpressionTree) tree;
+        currentBlock.elements.add(aaet);
+        build(aaet.expression());
+        build(aaet.dimension());
+        break;
+      }
+      case ARRAY_DIMENSION: {
+        ArrayDimensionTree arrayDimensionTree = (ArrayDimensionTree) tree;
+        if(arrayDimensionTree.expression() != null) {
+          build(arrayDimensionTree.expression());
+        }
+        break;
+      }
+      case NEW_CLASS: {
+        NewClassTree newClassTree = (NewClassTree) tree;
+        currentBlock.elements.add(newClassTree);
+        if(newClassTree.enclosingExpression() != null) {
+          build(newClassTree.enclosingExpression());
+        }
+        build(Lists.reverse(newClassTree.arguments()));
+      }
+      break;
+      case IDENTIFIER:
+      case INT_LITERAL:
+      case LONG_LITERAL:
+      case DOUBLE_LITERAL:
+      case CHAR_LITERAL:
+      case FLOAT_LITERAL:
+      case STRING_LITERAL:
+      case BOOLEAN_LITERAL:
+      case NULL_LITERAL:
+        currentBlock.elements.add(tree);
+        break;
+      case TYPE_CAST:
+        currentBlock.elements.add(tree);
+        TypeCastTree typeCastTree = (TypeCastTree) tree;
+        build(typeCastTree.expression());
+        break;
+      case INSTANCE_OF:
+        InstanceOfTree instanceOfTree = (InstanceOfTree) tree;
+        currentBlock.elements.add(instanceOfTree);
+        build(instanceOfTree.expression());
+        break;
+      case EMPTY_STATEMENT:
+        break;
+      case NEW_ARRAY:
+        NewArrayTree newArrayTree = (NewArrayTree) tree;
+        currentBlock.elements.add(tree);
+        build(Lists.reverse(newArrayTree.dimensions()));
+        build(Lists.reverse(newArrayTree.initializers()));
+        break;
+      case CLASS:
+      case ENUM:
+      case ANNOTATION_TYPE:
+      case INTERFACE:
+        //Ignore declarations
+        break;
+      //Java 8 constructions : ignored for now.
+      case METHOD_REFERENCE:
+      case LAMBDA_EXPRESSION:
+        break;
+      default:
+        throw new UnsupportedOperationException(tree.kind().name()+ " " + ((JavaTree) tree).getLine());
+    }
+    }
+
+  private Block createUnconditionalJump(Tree terminator, @Nullable Block target) {
+    Block result = createBlock();
+    result.terminator = terminator;
+    if (target != null) {
+      result.successors.add(target);
+    }
+    return result;
+  }
+
+  private void buildCondition(Tree syntaxNode, Block trueBlock, Block falseBlock) {
+    switch (syntaxNode.kind()) {
+      case CONDITIONAL_OR: {
+        BinaryExpressionTree e = (BinaryExpressionTree) syntaxNode;
+        // process RHS
+        buildCondition(e.rightOperand(), trueBlock, falseBlock);
+        falseBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        buildCondition(e.leftOperand(), trueBlock, falseBlock);
+        break;
+      }
+      case CONDITIONAL_AND: {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 11 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1151)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 9. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1199)